### PR TITLE
Improved Datadog integration

### PR DESCRIPTION
**Pull request by markbastiaans**
This PR includes the following changes to the Datadog and Telegraf integrations:

- Use Telegraf agent for collecting PostgreSQL metrics and database diskstorage size metric. Telegraf will send these metrics directly to Datadog if Datadog is enabled.
- Use Datadog IoT and trace agents for collecting application metrics, JMX metrics, logs and traces
- Use official Datadog Cloud Foundry buildpack. This saves 400+ MB of application disk space and makes the integration more maintainable in the future.
- Added logs redaction - email addresses will be redacted by default. Can be switched off by setting the `DATADOG_LOGS_REDACTION` environment variable to `false`.
- Removed direct Databroker dependencies in Datadog module
- Refactoring and improvements: tags, configuration templates and more

### Known Limitations
Telegraf [does not support (Datadog) metric types correctly yet](https://github.com/influxdata/telegraf/issues/6822) (e.g. rate, counter, gauge). This means that all database metrics are currently pushed to Datadog as a gauge.

The most important metrics (`before_xid_wraparound`, `connections`, `database_size`, `db.count`, `locks`, `max_connections`, `percent_usage_connections`, `table.count`, `deadlocks`) are gauges and are compatible with the Datadog PostgreSQL integration and associated dashboards.

*If you do require the additional rate and counter metrics, there is a workaround available.* First, set the `DATADOG_DATABASE_RATE_COUNT_METRICS` environment variable to `true`. After that variable is enabled, the rate and counter metrics are suffixed by either `_count` or `_rate` to prevent collisions with the official Datadog metrics. In the Datadog UI, the [metric type and unit can be changed](https://docs.datadoghq.com/developers/metrics/type_modifiers/?tab=count#modify-a-metrics-type-within-datadog) to reflect this. We also set a helpful `interval` tag (`10s`) which can be used here. Additionally, gauge metrics can be [rolled up in Datadog dashboards](https://docs.datadoghq.com/dashboards/functions/rollup/). The correct type and unit for other submitted metrics can be found [here](https://github.com/DataDog/integrations-core/blob/master/postgres/metadata.csv).

### CODE REVIEW

**Review by carbone-kid — APPROVED** [review]

File: buildpack/databroker/__init__.py [hunk 1]
```diff
@@ -84,7 +90,55 @@ def __setup_configs(self, database_config):
...
+                    kafka_streams_cfg_path, kafka_streams_cfg,
+                )
+                jmx_config_files = [
+                    kafka_connect_cfg_path,
+                    kafka_streams_cfg_path,
+                ]
+            else:
+                # consumer metrics setup
+                extra_jmx_instance_config = consumer.jmx_metrics
+            update_config(
```
Comment (markbastiaans) [inline buildpack/databroker/__init__.py:134]: @abhishekamte-mx - what happens if you call this method twice?

**Review by matyaskuti — COMMENTED** [review]
Looks okay in general, a few small comments and questions, nothing big related to the overall goal.

File: buildpack/databroker/__init__.py [hunk 2]
```diff
@@ -84,7 +89,48 @@ def __setup_configs(self, database_config):
...
         return complete_conf
 
-    def run(self, m2ee_client, database_config):
+    def get_datadog_config(self, user_checks_dir):
+        extra_jmx_instance_config = None
+        jmx_config_files = []
+        if is_enabled():
+            if self.is_producer_app:
+                # kafka connect cfg
+                os.makedirs(
```
[the remark this answers is not in the document]
Comment (markbastiaans) [reply buildpack/databroker/__init__.py:98]: I agree that it has inconsistent use, but `os.makedirs` is the recommended way to do this nowadays.

File: buildpack/datadog.py [hunk 3]
```diff
@@ -290,41 +312,60 @@ def _set_up_environment():
...
+    if _is_logs_redaction_enabled():
+        logging.info(
+            "Datadog logs redaction enabled, all email addresses will be redacted"
+        )
+        log_processing_rules = {
+            "log_processing_rules": [
+                {
+                    "type": "mask_sequences",
+                    "name": "RFC_5322_email",
+                    "pattern": r"""(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])""",
```
[the remark this answers is not in the document]
Comment (markbastiaans) [reply buildpack/datadog.py:356]: Yes, and from the Datadog documentation.

File: tests/unit/test_datadog.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import os
+import json
+
+from buildpack import datadog
+
+
+class TestCaseDatadogUtilFunctions:
```
[the remark this answers is not in the document]
Comment (markbastiaans) [reply tests/unit/test_datadog.py:7]: I agree, and that would be something to apply to almost all unit tests in the buildpack. Can we keep it out-of-scope for this improvement?

[the remark this answers is not in the document]
Comment (matyaskuti) [reply tests/unit/test_datadog.py:7]: Applying it to the rest of the tests, sure, we can skip that. But as this is being introduced here, I'd prefer it done idiomatically here.

**Review by matyaskuti — APPROVED** [review]

**Pull request merged**